pingcap/tidb-operator
# Use GenericConfic for pd/tidb/tikv/tiflash/ticdc

### What problem does this PR solve? 

Allow passing what's specified by the user and pass them to tidb/tikv/pd/tiflash as toml config file

#### Know issue
For pd, this config item may have a compatibility problem.
It Accepted "true" and "false" string literal before.
After this PR, we will pass it as a string literal to pd if user still doing this.
<img width="546" alt="Screen Shot 2020-09-16 at 7 08 44 PM" src="https://user-images.githubusercontent.com/1681864/93329665-1b75d280-f850-11ea-8edd-6f231606556c.png">


### What is changed and how does it work?

Use GenericConfic for tidb/tikv/tiflash/ticdc

### Check List 

Tests 


 - Manual test (add detailed scripts or steps below)
deploy a cluster set a config item to tidb
```
  tidb:
    baseImage: pingcap/tidb
    replicas: 1
    service:
      type: ClusterIP
    config:
      what-the: "ok"
```
check logs of tidb:
<img width="667" alt="Screen Shot 2020-09-16 at 8 14 04 PM" src="https://user-images.githubusercontent.com/1681864/93335890-a60eff80-f859-11ea-84eb-1c2524c6c8df.png">
tidb will just print wrong log for unknow item default.

Code changes

 - Has Go code change

Side effects

 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch

### Does this PR introduce a user-facing change?:

```release-note
Allow passing what's specified by the user and pass them to tidb/tikv/pd/tiflash as toml config file
```
